cclient/iota_client_extended_api.c

```diff
@@ -546,3 +645,148 @@ retcode_t iota_client_send_transfer(
+  if (!broadcast_transactions_req) {
+    ret_code = RC_CCLIENT_NULL_PTR;
+    log_error(logger_id, "%s broadcast_transactions_req_new failed: %s\n",
+              __func__, error_2_string(ret_code));
+    goto cleanup;
+  }
+
+  log_debug(logger_id, "[%s:%d]\n", __func__, __LINE__);
+  ret_code = traverse_bundle(serv, tail_hash, bundle,
+                             broadcast_transactions_req->trytes);
+  if (ret_code != RC_OK) {
+    log_error(logger_id, "%s traverse_bundle failed: %s\n", __func__,
+              error_2_string(ret_code));
+    goto cleanup;
+  }
+  // Call the broadcast_transactions API
+  ret_code =
+      iota_client_broadcast_transactions(serv, broadcast_transactions_req);
+  if (ret_code != RC_OK) {
+    log_error(logger_id, "%s iota_client_broadcast_transactions failed: %s\n",
+              __func__, error_2_string(ret_code));
+    goto cleanup;
+  }
+
+cleanup:
+  broadcast_transactions_req_free(&broadcast_transactions_req);
+  return ret_code;
+}
+
+bundle_status_t iota_client_get_bundle(iota_client_service_t const* const serv,
+                                       flex_trit_t const* const tail_hash,
+                                       bundle_transactions_t* const bundle) {
+  bundle_status_t bundle_status = BUNDLE_NOT_INITIALIZED;
+  log_debug(logger_id, "[%s:%d]\n", __func__, __LINE__);
+  if (iota_client_traverse_bundle(serv, tail_hash, bundle) == RC_OK) {
+    bundle_validator(bundle, &bundle_status);
+  }
+  return bundle_status;
+}
+
+retcode_t iota_client_traverse_bundle(iota_client_service_t const* const serv,
+                                      flex_trit_t const* const tail_hash,
+                                      bundle_transactions_t* const bundle) {
+  log_debug(logger_id, "[%s:%d]\n", __func__, __LINE__);
+
+  return traverse_bundle(serv, tail_hash, bundle, NULL);
+}
+
+retcode_t iota_client_promote_transaction(
+    iota_client_service_t const* const serv, flex_trit_t const* const tail_hash,
+    int const depth, int const mwm, transfer_list_t const* const transfers,
+    transaction_array_t* const tx_objs) {
+  log_debug(logger_id, "[%s:%d]\n", __func__, __LINE__);
+  retcode_t ret_code = RC_OK;
+  iota_transaction_t* spam_transaction = NULL;
+
+  // check consistency
+  check_consistency_req_t* consistency_req = check_consistency_req_new();
+  check_consistency_res_t* consistency_res = check_consistency_res_new();
+  if (!consistency_req || !consistency_res) {
+    ret_code = RC_CCLIENT_OOM;
+    goto done;
+  }
+
+  hash243_queue_push(&consistency_req->tails, tail_hash);
+  ret_code =
+      iota_client_check_consistency(serv, consistency_req, consistency_res);
+  if (ret_code != RC_OK) {
+    log_error(logger_id, "%s checking consistency failed: %s\n", __func__,
+              error_2_string(ret_code));
+    goto done;
+  }
+
+  if (consistency_res->state) {
+    // adding spam transaction
+    spam_transaction = transaction_new();
```

Comment:
Just noting for the next PR that this need to be freed

Reply:
right, thanks.